src/pixel-ping.coffee

```diff
@@ -45,15 +45,20 @@ merge = (new_store) ->
 flush = ->
   log store
   return unless config.endpoint
+  on_error = (message) ->
+    if !config.discard
+      merge(old_store)
+    console.log message
   data = serialize()
   old_store = reset()
   endReqOpts['headers']['Content-Length'] = data.length
-  request = http.request endReqOpts, (response) ->
-    console.info '--- flushed ---'
+  request = http.request endReqOpts, (res) ->
+    if res.statusCode <= 299 and res.statusCode >= 200
+      console.info '--- flushed ---'
+    else
+      on_error "--- flush failed with code:" + res.statusCode
   request.on 'error', (e) ->
-    if !config.discard
-      merge(old_store)
-    console.log "--- cannot connect to endpoint : #{e.message}"
+    on_error "--- cannot connect to endpoint : #{e.message}"
```

Comment:
Should `400` codes list `cannot connect`? Aren't they just generic application errors? Maybe would should `switch` out the response codes into classes?

Reply:
You're right, they shouldn't. I'm pretty sure the error event only happens on a network or protocol error.

From [the docs](https://nodejs.org/dist/latest-v6.x/docs/api/http.html):

> If any error is encountered during the request (be that with DNS resolution, TCP level errors, or actual HTTP parse errors) an 'error' event is emitted on the returned request object. As with all 'error' events, if no listeners are registered the error will be thrown.